analyses/cell-type-ewings/template_notebooks/auc-singler-workflow/03-clustering.Rmd

```diff
@@ -0,0 +1,265 @@
+- We then look at marker gene expression across the clusters and assign each cluster to a cell type. 
+
+## Setup
+
+```{r}
+# check that sce and results files exist
+stopifnot(
+  "sce file does not exist" = file.exists(params$sce_file),
+  "SingleR results file does not exist" = file.exists(params$singler_results_file),
+  "Marker genes file does not exist" = file.exists(params$marker_genes_file)
+)
+```
+
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+
+# Set default ggplot theme
+theme_set(
+  theme_classic()
+)
+
+# quiet messages
+options(readr.show_col_types = FALSE)
+ComplexHeatmap::ht_opt(message = FALSE)
+```
+
+
+```{r base paths}
+# The path to this module
+module_base <- rprojroot::find_root(rprojroot::is_renv_project)
+
+# source in helper functions: calculate_sum_markers()
+validation_functions <- file.path(module_base, "scripts", "utils", "tumor-validation-helpers.R")
+source(validation_functions)
+
+# source in clustering functions
+clustering_functions <- file.path(module_base, "scripts", "utils", "clustering-functions.R")
+source(clustering_functions)
+```
+
+
+```{r}
+# read in input sce files
+sce <- readr::read_rds(params$sce_file)
+
+# read in singler results
+singler_classification_df <- readr::read_tsv(params$singler_results_file)
+```
+
+## Clustering
+
+Below we perform Louvain, Jaccard clustering, varying `k`. 
+The minimum `k` is 5 and the maximum `k` is 40 with a step size of 5. 
+
+```{r}
+# perform clustering across k = 5-40 with increments of 5
+all_cluster_results <- cluster_sweep(sce) |>
+  dplyr::mutate(
+    nn_param = forcats::fct_relevel(nn_param, "k_5", after = 0)
+  )
+```
+
+```{r}
+# get umap embeddings and combine into a data frame with cluster assignments
+umap_df <- sce |>
+  scuttle::makePerCellDF(use.dimred = "UMAP") |>
+  # replace UMAP.1 with UMAP1 and get rid of excess columns
+  dplyr::select(barcodes, UMAP1 = UMAP.1, UMAP2 = UMAP.2) |>
+  dplyr::left_join(all_cluster_results, by = "barcodes")
+```
+
+
+Below we visualize the cluster assignments using each parameter on a UMAP. 
+This can be helpful to see any values of K that may have obvious over or under clustering. 
+
+```{r, fig.height=15, fig.width=10}
+# look at clustering results for each library across all params
+ggplot(umap_df, aes(x = UMAP1, y = UMAP2, color = cluster)) +
+  geom_point(alpha = 0.5, size = 0.1) +
+  facet_wrap(vars(nn_param)) +
+  theme(
+    aspect.ratio = 1
+  ) +
+  guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5)))
+```
+
+## Clustering statistics 
+
+Below we calculate a series of statistics: 
+
+- Average silhouette width: This metric evaluates cluster separation. 
+Cells with large positive silhouette widths are closer to other cells in the same cluster than to cells in different clusters. 
```

Comment:
I'd add a "higher = ..." sentence here. Not convinced I have the right wording here, but along these lines.

```suggestion
Cells with large positive silhouette widths are closer to other cells in the same cluster than to cells in different clusters. 
Higher values indicate tighter clusters.
```